mlir/lib/Dialect/Affine/Transforms/AffineStoreAndLoad.cpp

```diff
@@ -0,0 +1,73 @@
+#include "PassDetail.h"
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "mlir/Dialect/Affine/Passes.h"
+#include "llvm/Support/Debug.h"
+
+#define DEBUG_TYPE "affine-store"
+
+using namespace mlir;
+
+namespace {
+struct AffineStorePass : public AffineStoreBase<AffineStorePass> {
+  void runOnFunction() override;
+};
+struct AffineLoadPass : public AffineLoadBase<AffineLoadPass> {
+  void runOnFunction() override;
+};
+} // namespace
+
+static bool isValidIndex(Value index) {
+  if (index.isa<BlockArgument>() &&
+      isa<AffineForOp>(index.cast<BlockArgument>().getOwner()->getParentOp()))
+    return true;
```

Comment:
I'm not sure this is sufficient. What if you have load x[i+1]? Specifically an add, mul, sub is also valid

Reply:
Yes, that's correct. I will handle x[i+something] in the next commit. I am finishing writing the test cases and then I will create a PR.

Reply:
Sounds good, feel free to merge!